Mutagen.Bethesda.Analyzers.Engine/Drivers/IsolatedDriverParams.cs

```diff
@@ -10,20 +11,29 @@ public class IsolatedDriverParams
     public readonly ILinkCache LinkCache;
     public readonly IReportDropbox ReportDropbox;
     public readonly IModGetter TargetMod;
-    public readonly ModPath TargetModPath;
+    public readonly IsolatedDriverFileParams? FileParams;
     public readonly CancellationToken CancellationToken;
 
+
     public IsolatedDriverParams(
         ILinkCache linkCache,
         IReportDropbox reportDropbox,
         IModGetter targetMod,
-        ModPath modPath,
+        IsolatedDriverFileParams? fileParams,
         CancellationToken cancellationToken)
     {
         LinkCache = linkCache;
         ReportDropbox = reportDropbox;
         TargetMod = targetMod;
-        TargetModPath = modPath;
+        FileParams = fileParams;
         CancellationToken = cancellationToken;
     }
 }
+
+public class IsolatedDriverFileParams(
```

Comment:
Next to be able to have a custom file system, it might also be relevant to be able to pass in mod paths with different data directories. Currently you can input a custom data directory and run it all from there, but I for example also have a use case where I get mods from various MO2 folders which I all put into a game environment. This use case wouldn't be supported atm.

That would mean a mod path which is a data relative path isn't sufficient to locate a mod anymore, and you'd need to be able to target all mods individually. Probably something we don't need for the first release but it would be good to offer support for this in the future? Maybe something for an issue?

Reply:
Hmm.  The only reason Data folder is needed at all is for the RecordFrame analyzers, which want to open a fresh stream to the mod's file.   They use the Data folder + ModKey to find the mod file to read in.

We could delete the Data folder concept entirely, and have them find out the mod paths a different way

Let's maybe do as a followup PR just to get this one in

Reply:
A followup PR should be fine, yeah! Do we open an issue for that?